src/js_results.hpp

```diff
@@ -302,7 +304,31 @@ void ResultsClass<T>::index_of(ContextType ctx, FunctionType, ObjectType this_ob
         return_value.set(-1);
     }
 }
-    
+
+template<typename T>
+void ResultsClass<T>::update(ContextType ctx, FunctionType, ObjectType this_object, size_t argc, const ValueType arguments[], ReturnValue &return_value) {
+    validate_argument_count(argc, 2);
+
+    std::string property = Value::validated_to_string(ctx, arguments[0], "property");
+    auto results = get_internal<T, ResultsClass<T>>(this_object);
+
+    auto schema = results->get_object_schema();
+    if (!schema.property_for_name(StringData(property))) {
+        throw std::invalid_argument(util::format("No such property: %1", property));
+    }
+
+    auto realm = results->get_realm();
+    if (!realm->is_in_transaction()) {
+        throw std::runtime_error("Can only 'update' objects within a transaction.");
+    }
+
+    for (auto i = results->size(); i > 0; i--) {
```

Comment:
I am missing something here. How this is different from doing it in JS like 
```
var results = realm.objects('NullableBasicTypesObject');
         realm.write(() => {
                 for (k i n results)
                      k.intCol = 20;
         });
```
minus the JS transitions.

can't we do a more optimized version that goes directly into object store so we gain from bulk updates

Reply:
It's just moving the loop to C++ for now, with some performance gains. Ideally yes, this would be implemented in OS or Core in a "native" manner for much better performance, but using this approach to keep things simple to start with.

If I'm not mistaken, @bmunkholm was alluding to the same in his last comment. I'm going to leave a TODO to move it to a lower layer.

Reply:
A optimization for the future is to operate directly on `TableView`. But that might be better done in object store than here.

Reply:
I think we don't talk about the same thing as @bmunkholm 
My reasoning is that this creates instances of RealmObjectClass<T> just to call set_property on it which I think is the same as doing it in JS except the JS transitions. What I am wondering is could we possible refactor `set_property` or user NativeAccessor directly to implement this. I think cocoa is doing something similar. Maybe I am over engineering this but the cocoa approach seems much more optimized

Reply:
here is what cocoa is doing

https://github.com/realm/realm-cocoa/blob/9df4bcb39459bb58a376a4ef0e088f8fc2376a83/Realm/RLMCollection.mm#L231

Reply:
I think @bmunkholm  suggest we have multiple property changes in one call which currently can be done in JS like so
```
results.update({ prop1: val1, prop2: val2, prop3: val3 })
(for  updatePair in arguments )
(pair => results.update(updatePair))
```
which is a "convenience method" if done like so in JS
but which can be optimized if core provided us with an `updateMany` method that will be a single call in realm-js as well

Reply:
@blagoev I see your point. Will investigate how to avoid creating `RealmObjectClass` instances inside the loop, using `TableView` directly like Cocoa.